torch/nn/modules/transformer.py

```diff
@@ -92,11 +92,12 @@ def forward(self, src, tgt, src_mask=None, tgt_mask=None,
             Note: [src/tgt/memory]_mask should be filled with
             float('-inf') for the masked positions and float(0.0) else. These masks
             ensure that predictions for position i depend only on the unmasked positions
-            j and are applied identically for each sequence in a batch.
-            [src/tgt/memory]_key_padding_mask should be a ByteTensor where True values are positions
-            that should be masked with float('-inf') and False values will be unchanged.
-            This mask ensures that no information will be taken from position i if
-            it is masked, and has a separate mask for each sequence in a batch.
+            j and are applied identically for each sequence in a batch. If a ByteTensor is provided,
```

Comment:
I wouldn't talk about this in terms of describing the code that implements this extended semantics but describing the semantics itself. What if we at some point can avoid the fill operation? The semantics won't change, but we'll need to change all these docs. If someone wants to know these details they should look at the code.